docs/service-layer/other-events.md

```diff
@@ -0,0 +1,192 @@
+# Index
+
+- [Retrieve New Messages](#Retrieve-Messages)
+- [Delete Messages](#Delete-Messages)
+- [Retrieve DID](#Retrieve-DID)
+- [Delete Contact](#Delete-Contact)
+- [Register Contact](#Register-Contact)
+
+# Retrieve New Messages
+
+This event will bring new messages to the FE, which will be stored in the local storage. There will be a limit on cached messages. The limit will be per conversation. Once there are new messages that exceed the pile, we will delete old ones.
+
+## Example logic
+
+```javascript
+if (cachedMessages.length > maxMessages) {
+  const prunedMessages = cachedMessages.slice(-maxMessages);
+  localStorage.setItem(key, JSON.stringify(prunedMessages));
+}
+```
+
+## Example input:
+
+This method receives no input.
+
+## Example response:
+
+```json
+{
+  "status": "success",
+  "payload": [
+    {
+      "did": "did:peer(..)",
+      "contact": "Alice",
+      "message": {"Hey Peter, bring bread."},
+      "id": 3
+    },
+    {
+      "did": "did:peer(..)",
+      "contact": "Bob",
+      "message": {"Hey Bob, bring butter."},
+      "id": 2
+    },
+    {
+      "did": "did:peer(..)",
+      "contact": "Hugo",
+      "message": {"Hey Jan, bring Salt."},
+      "id": 1
+    }
+  ]
+}
+```
+
+## Event channel name:
+
+```bash
+RetrieveMessages
+```
+
+# Delete Messages
+
+This event will delete specific messages.
+
+## Example input:
+
+```javascript
+DeleteMessages(did, id);
+DeleteMessages('did:peer', 3);
+```
```

Comment:
Aren't message IDs unique across the collection?

Reply:
True, we can use only the message id.